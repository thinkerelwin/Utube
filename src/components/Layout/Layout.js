import React from 'react';
import './Layout.scss';
import Header from '../../containers/Header/Header';

export function Layout(props) {
  return (
    <div className="app-layout">
      <Header />
      {props.children}
    </div>
  );
}
