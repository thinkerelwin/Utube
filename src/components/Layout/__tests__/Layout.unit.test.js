import React from 'react';
import { shallow } from 'enzyme';
import { Layout } from '../Layout';

test('renders empty <Layout/>', () => {
  const wrapper = shallow(<Layout />);
  expect(wrapper).toMatchSnapshot();
});

test('renders <Layout/> with one child', () => {
  const wrapper = shallow(
    <Layout>
      <div>Child 1</div>
    </Layout>
  );
  expect(wrapper).toMatchSnapshot();
});

test('renders <Layout/> with children', () => {
  const wrapper = shallow(
    <Layout>
      <div>Child</div>
      <div>
        <span>Child</span>
        <p>Child</p>
      </div>
    </Layout>
  );
  expect(wrapper).toMatchSnapshot();
});
