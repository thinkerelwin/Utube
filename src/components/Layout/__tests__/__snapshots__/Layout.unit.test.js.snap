// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Layout/> with children 1`] = `
<div
  className="app-layout"
>
  <Header />
  <div>
    Child
  </div>
  <div>
    <span>
      Child
    </span>
    <p>
      Child
    </p>
  </div>
</div>
`;

exports[`renders <Layout/> with one child 1`] = `
<div
  className="app-layout"
>
  <Header />
  <div>
    Child 1
  </div>
</div>
`;

exports[`renders empty <Layout/> 1`] = `
<div
  className="app-layout"
>
  <Header />
</div>
`;
